### PR TITLE
adapter: add metrics for some adapter client ops

### PR DESCRIPTION
**Pull request by mjibson**
While here, small refactor to cleanup common use of cmd_tx.send().

See https://github.com/MaterializeInc/cloud/issues/4146

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: src/adapter/src/client.rs [hunk 1]
```diff
@@ -53,6 +56,23 @@ impl Handle {
...
+struct Metrics {
+    pub commands: IntCounterVec,
+}
+
+impl Metrics {
+    pub fn register_into(registry: &MetricsRegistry) -> Metrics {
+        Metrics {
+            commands: registry.register(metric!(
+                name: "mz_adapter_commands",
+                help: "The total number of adapter commands issued  of the given type since process start.",
```
Comment (jkosh44) [inline src/adapter/src/client.rs:69]: Not a huge deal but it looks like there might be two spaces in between `issued` and `of`.

**Review by jkosh44 — APPROVED** [review]
LGTM

**Review by doy-materialize — APPROVED** [review]
thanks!

File: src/adapter/src/client.rs (hunk 1 above)
Comment (mjibson) [reply src/adapter/src/client.rs]: Fixed.

**Pull request merged**